### PR TITLE
perforce: allow configuration of p4.client

### PR DESCRIPTION
**Pull request by bobheadxi**
Add `p4.client` to Perforce configuration, which will then be used to imply the `--use-client-spec` option added in #23833 

Follow-up to https://github.com/sourcegraph/sourcegraph/pull/23833, https://github.com/sourcegraph/customer/issues/445

### CODE REVIEW

File: cmd/gitserver/main.go [hunk 1]
```diff
@@ -137,7 +137,8 @@ func main() {
 
 				return &server.PerforceDepotSyncer{
 					MaxChanges:    int(c.MaxChanges),
-					UseClientSpec: c.UseClientSpec,
+					UseClientSpec: c.P4Client != "",
```
Comment (unknwon) [inline cmd/gitserver/main.go:140]: Looks like we don't need this field anymore.

Comment (bobheadxi) [reply cmd/gitserver/main.go:140]: I was thinking that `PerforceDepotSyncer` can be the lowest-level implementation that reflects the command line options exactly, but I can remove this too!

Comment (bobheadxi) [reply cmd/gitserver/main.go]: Removed in https://github.com/sourcegraph/sourcegraph/pull/23845/commits/43f9681135d6d4dbbc659468f6a329a942e80d67

**Review by unknwon — APPROVED** [review]
Thanks!

**Pull request merged**